ebl-issuance/src/main/java/org/dcsa/conformance/standards/eblissuance/party/EblIssuanceCarrier.java

```diff
@@ -190,6 +197,40 @@ private void sendIssuanceRequest(JsonNode actionPrompt) {
             .formatted(isCorrect ? "correct" : "incorrect", tdr, eblStatesByTdr.get(tdr)));
   }
 
+  private ObjectNode getSupportingDocumentObject(){
+    var document = generateDocument();
+    return OBJECT_MAPPER.createObjectNode()
+    .put("name","test-iss-document")
+      .put("content",document)
+      .put("mediatype","application/octet-stream");
+  }
+
+  @SneakyThrows
+  private static byte[] generateDocument() {
+    var uuid = UUID.randomUUID();
+    byte[] pdf;
+    String filepath = "/standards/eblissuance/messages/test-iss-document.pdf";
+    try (InputStream inputStream = EblIssuanceCarrier.class.getResourceAsStream(filepath)) {
+      if (inputStream == null) {
+        throw new IllegalArgumentException("Could not resolve " + filepath);
+      }
+      pdf = inputStream.readAllBytes();
+    }
+    var docWithUuid = new byte[pdf.length + 16];
+
+    // Add a UUID to ensure it is unique, such that the testers do not have to "remove" the
+    // document between every test.
+    System.arraycopy(pdf, 0, docWithUuid, 0, pdf.length);
+    putLong(uuid.getMostSignificantBits(), docWithUuid, pdf.length);
+    putLong(uuid.getLeastSignificantBits(), docWithUuid, pdf.length + 8);
+    return docWithUuid;
+  }
```

Comment:
Some minor remarks: 

1. Line 210 has a variable declaration, but you only use it at 224. Please move it down, directly before you need it, so it is clear why it is declared. 
2. I wonder if you remove line 214-216, does that not already throw an exception if the file is not there? Can you try? Because `inputStream.readAllBytes()` doesn't fail secretly, I guess. If that does not throw a clear exception, you can catch an exception after the `try (....) { .... } catch(` and throw the exception with added text as well. For example `throw new IOException("Could not read the PDF file. " + e.getMessage());`
3. Regarding Niels' question: I don't know if that matters, but if it matters, you can change the code a bit on your laptop, to save the same file temporarily to a new file, and try to open it locally.